app/code/core/Mage/Adminhtml/Block/Widget/Grid.php

```diff
@@ -1886,4 +1893,29 @@ public function getLimitOptions(): array
     {
         return [20, 30, 50, 100, 200, 500, 1000];
     }
+
+    /**
+     * @param string $aclPath ACL path
+     * @return bool
+     */
+    protected function canView(string $aclPath): bool
+    {
+        return $this->_isAllowed($aclPath);
+    }
+
+    /**
+     * Cache whether grid row is accessible
+     * @param string $aclPath ACL path
+     * @return bool
+     */
+    private function _isAllowed(string $aclPath): bool
+    {
+        if ($this->isAllowed === null || !isset($this->isAllowed[$aclPath])) {
+            /** @var Mage_Admin_Model_Session $session */
+            $session = Mage::getSingleton('admin/session');
+            $this->isAllowed[$aclPath] = $session->isAllowed($aclPath);
+        }
+
+        return $this->isAllowed[$aclPath];
```

Comment:
if the array was always an array

```suggestion
        if (!isset($this->isAllowed[$aclPath]) {
            /** @var Mage_Admin_Model_Session $session */
            $session = Mage::getSingleton('admin/session');
            $this->isAllowed[$aclPath] = $session->isAllowed($aclPath);
        }

        return $this->isAllowed[$aclPath];
```
?